phoenix-core/src/main/java/org/apache/phoenix/index/IndexMaintainer.java

```diff
@@ -1030,35 +1032,40 @@ public Put buildUpdateMutation(KeyValueBuilder kvBuilder, ValueGetter valueGette
                 for (Pair<ColumnReference, ColumnReference> colRefPair : colRefPairs) {
                     ColumnReference indexColRef = colRefPair.getFirst();
                     ColumnReference dataColRef = colRefPair.getSecond();
-                    Expression expression = new SingleCellColumnExpression(new PDatum() {
-                        @Override
-                        public boolean isNullable() {
-                            return false;
-                        }
-                        
-                        @Override
-                        public SortOrder getSortOrder() {
-                            return null;
-                        }
-                        
-                        @Override
-                        public Integer getScale() {
-                            return null;
-                        }
-                        
-                        @Override
-                        public Integer getMaxLength() {
-                            return null;
-                        }
-                        
-                        @Override
-                        public PDataType getDataType() {
-                            return null;
+
+                    byte[] value = null;
+                    if (this.dataTableImmutableStorageScheme == this.immutableStorageScheme) {
```

Comment:
It is not clear to me how this extra check is useful here and how it fixes the issue. Why is this check necessary, given that the index is supposed to  inherit the storage scheme from the data table currently? Is it possible that they can be different.